Application/Telegram/TelegramCommunicationService.cs

```diff
@@ -34,16 +34,19 @@ public async Task Execute(Update update)
             {
                 if (UserStateStorage.GetUserCurrentState(update.Message.From.Id) == UserState.ProcessNotStarted)
                 {
-                    UserStateStorage.AddUser(5213829376, UserState.ProcessNotStarted, UserRole.User);
+                    UserStateStorage.AddUser(5213829376, UserState.ProcessNotStarted, UserRole.User );
+                    UserStateStorage.AddRecordToUserMessages(5213829376, new List<int>());
                 }
             }
-            
-            //await _telegraBotClient.SendChatActionAsync(update.Message.Chat.Id, ChatAction.Typing);
+                       
             switch (update.Type)
             {
                 case UpdateType.CallbackQuery:
-                    await _telegraBotClient.SendTextMessageAsync(update.CallbackQuery.Message.Chat.Id,
-                                                $"Button pressed at {DateTime.Now.ToString("h: mm:ss tt")} \n");
+                    if (update.CallbackQuery.Data.Contains("BACK"))
+                    {                        
+                        UserStateStorage.UserStateUpdate(update.CallbackQuery.From.Id,
+                            (UserState)Enum.Parse(typeof(UserState), update.CallbackQuery.Data.Substring(4)));
```

Comment:
why substring?

Reply:
because in BACK button call back data I'm putting userstate in this format: "BACKUserState", so substring(4) gives me userstate.